src/serialization/markup.hpp

```diff
@@ -0,0 +1,111 @@
+/*
+	Copyright (C) 2024
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "color.hpp"
+
+// This file isn't needed by any of these functions, but this allows any
+// standard color to be passed to span_color without an extra include.
+#include "font/standard_colors.hpp"
+#include "formatter.hpp"
+#include "serialization/string_utils.hpp"
+
+#include <string>
+
+namespace markup {
+
+/**
+ * Returns the contents enclosed inside &lt;tag_name&gt; and &lt;/tag_name&gt;
+ */
+template<typename... Args>
+std::string tag(const std::string& tag_name, Args... contents)
+{
+	return ((formatter()
+		<< "<" << tag_name << ">")
+		<< ...
+		<< contents)
+		<< "</" << tag_name << ">";
+}
+
+template<typename... Args>
+std::string tag_esc(const std::string& tag_name, Args... contents)
+{
+	return ((formatter()
+		<< "<" << tag_name << ">")
+		<< ...
+		<< utils::escape(contents, "'\\"))
+		<< "</" << tag_name << ">";
+}
+
+/**
+ * Returns a Pango formatting string using the provided color_t object.
+ *
+ * The string returned will be in format: &lt;span foreground=\#color$gt;
+ * Callers will need to manually append the closing &lt;/span$gt;' tag.
+ *
+ * @param color        The color_t object from which to retrieve the color.
+ */
+std::string span_color(const color_t& color);
```

Comment:
How many places use this overload? It should optimally be refactored out.

Reply:
going in that direction.

Reply:
removed.